validator/rules/cmd_test.go

```diff
@@ -0,0 +1,104 @@
+package rules
+
+import (
+	"testing"
+
+	"github.com/aerogear/charmil/validator"
+	"github.com/spf13/cobra"
+)
+
+func Test_ExecuteCommand(t *testing.T) {
+	cmd := rootCmd
+
+	// Testing cobra commands with default recommended config
+	// default config can also be overrided
+	ruleCfg := ValidatorConfig{
+		ValidatorOptions: ValidatorOptions{
+			SkipCommands: map[string]bool{"cmd100 cmd0 subcmd01": true},
+		},
+		ValidatorRules: ValidatorRules{
+			Length: Length{
+				RuleOptions: validator.RuleOptions{
+					SkipCommands: map[string]bool{"cmd100": true},
+				},
+				Limits: map[string]Limit{
+					"Use":     {Min: 1},
+					"Example": {Min: 3},
+					"Long":    {Min: 5},
+				},
+			},
+			MustExist: MustExist{
+				RuleOptions: validator.RuleOptions{
+					SkipCommands: map[string]bool{"cmd100": true},
+				},
+				Fields: map[string]bool{"Run": true},
+			},
+			UseMatches: UseMatches{Regexp: `^[^-_+]+$`},
+		},
+	}
+
+	validationErr := ExecuteRules(cmd, &ruleCfg)
+	if len(validationErr) != 0 {
+		t.Errorf("validationErr was not empty, got length: %d; want %d", len(validationErr), 0)
+	}
```

Comment:
I think we can leave this out, the validation already prints out how many errors occur and we hard code in the want value here anyway.